x-pack/legacy/plugins/lens/public/editor_frame_service/editor_frame/config_panel_wrapper.tsx

```diff
@@ -174,85 +190,329 @@ function LayerPanels(
+              label={dimension.dimensionLabel}
+              key={index}
+              isInvalid={isMissing}
+              error={
+                isMissing
+                  ? i18n.translate('xpack.lens.editorFrame.requiredDimensionWarningLabel', {
+                      defaultMessage: 'Required dimension',
+                    })
+                  : []
+              }
+            >
+              <>
+                {dimension.accessors.map(accessor => (
+                  <DragDrop
+                    key={accessor}
+                    className="lnsIndexPatternDimensionPanel"
+                    data-test-subj="indexPattern-dropTarget"
+                    droppable={
+                      dragDropContext.dragging &&
+                      layerDatasource.canHandleDrop({
+                        ...layerDatasourceDropProps,
+                        columnId: accessor,
+                        filterOperations: dimension.filterOperations,
+                      })
+                    }
+                    onDrop={droppedItem => {
+                      layerDatasource.onDrop({
+                        ...layerDatasourceDropProps,
+                        droppedItem,
+                        columnId: accessor,
+                        filterOperations: dimension.filterOperations,
+                      });
+                    }}
+                  >
+                    {wrapInPopover(
+                      accessor,
+                      <NativeRenderer
+                        render={props.datasourceMap[datasourceId].renderDimensionTrigger}
+                        nativeProps={{
+                          ...layerDatasourceConfigProps,
+                          columnId: accessor,
+                          filterOperations: dimension.filterOperations,
+                          togglePopover: () => {
+                            if (popoverState.isOpen) {
+                              setPopoverState({
+                                isOpen: false,
+                                openId: null,
+                              });
+                            } else {
+                              setPopoverState({
+                                isOpen: true,
+                                openId: accessor,
+                              });
+                            }
+                          },
+                        }}
+                      />,
+                      <NativeRenderer
+                        render={props.datasourceMap[datasourceId].renderDimensionEditor}
+                        nativeProps={{
+                          ...layerDatasourceConfigProps,
+                          core: props.core,
+                          columnId: accessor,
+                          filterOperations: dimension.filterOperations,
+                        }}
+                      />
+                    )}
 
-      <EuiSpacer size="s" />
+                    <EuiButtonIcon
```

Comment:
For reviewers: **There is currently a bug here**. This function isn't actually removing the dimension from the datasource, it's only removed from the visualization.

Reply:
This bug is now fixed